go/runtime/history/indexer.go

```diff
@@ -289,6 +383,8 @@ func (bi *BlockIndexer) reindexRange(ctx context.Context, start int64, end int64
 		return err
 	}
 
+	bi.updateLastIndexedHeight(blocks)
```

Comment:
I would move this to `commitBlocks`, and then maybe just pass the last block to remove `if` check, or even better, remove this function and do that in the `commitBlocks`.